sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java

```diff
@@ -0,0 +1,299 @@
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.BaseSubscriber;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class AsynReadWithMultipleClients<T> {
+    private final static String PARTITION_KEY = "/pk";
+    private final Semaphore concurrencyControlSemaphore;
+    private final Logger logger;
+    private final Configuration configuration;
+    private MetricRegistry metricsRegistry = new MetricRegistry();
+    private ScheduledReporter reporter;
+    private Meter successMeter;
+    private Meter failureMeter;
+    private Timer latency;
+    private Map<CosmosAsyncClient, List<PojoizedJson>> clientDocsMap = new HashMap<>();
+
+    AsynReadWithMultipleClients(Configuration cfg) {
+        logger = LoggerFactory.getLogger(this.getClass());
+        this.configuration = cfg;
+        createClients();
+        if (configuration.getGraphiteEndpoint() != null) {
+            final Graphite graphite = new Graphite(new InetSocketAddress(configuration.getGraphiteEndpoint(), configuration.getGraphiteEndpointPort()));
+            reporter = GraphiteReporter.forRegistry(metricsRegistry)
+                .prefixedWith(configuration.getOperationType().name())
+                .convertRatesTo(TimeUnit.SECONDS)
+                .convertDurationsTo(TimeUnit.MILLISECONDS)
+                .filter(MetricFilter.ALL)
+                .build(graphite);
+        } else {
+            reporter = ConsoleReporter.forRegistry(metricsRegistry).convertRatesTo(TimeUnit.SECONDS)
+                .convertDurationsTo(TimeUnit.MILLISECONDS).build();
+        }
+        concurrencyControlSemaphore = new Semaphore(cfg.getConcurrency());
+    }
+
+    void deleteCollection() {
+        try {
+            for (CosmosAsyncClient cosmosAsyncClient : clientDocsMap.keySet()) {
+                cosmosAsyncClient.getDatabase(configuration.getDatabaseId()).getContainer(configuration.getCollectionId()).delete().block();
+            }
+        } catch (CosmosClientException e) {
+            if (e.getStatusCode() == HttpConstants.StatusCodes.NOTFOUND) {
+                // It can happen if container does not exist, normal scenario
+            } else {
+                throw e;
+            }
+        } finally {
+            System.out.println("Container on all client have been deleted successfully");
```

Comment:
please use logger, not System.out/err.
Please do everywhere.

Reply:
Done